ruby_event_store-rom/lib/ruby_event_store/rom/repositories/events.rb

```diff
@@ -51,6 +51,18 @@ def read(direction, stream, from:, limit:, batch_size:)
                 .to_ary
             end
             BatchEnumerator.new(batch_size, limit || Float::INFINITY, reader).each
+          elsif read_as == RubyEventStore::Specification::FIRST
+            stream_entries
+              .ordered(direction, stream, offset_entry_id)
+              .combine(:event)
+              .map_with(:stream_entry_to_serialized_record, auto_struct: false)
+              .to_ary.first
+          elsif read_as == RubyEventStore::Specification::LAST
+            stream_entries
+              .ordered(direction, stream, offset_entry_id)
+              .combine(:event)
+              .map_with(:stream_entry_to_serialized_record, auto_struct: false)
+              .to_ary.last
```

Comment:
@mpraglowski I think you don't want `.to_ary` here -- just `last` should get you one record. Internally, it reverses the SQL `ORDER` and `LIMIT`s to one record. I'm not sure if that means of reversing the query is always accurate. Presumably your tests check for that.

Reply:
however here `last` without `to_ary` results in failing tests:

```
     NoMethodError:
       undefined method `last' for #<ROM::Relation::Composite:0x00007fd4226cecf8>
     Shared Example Group: :event_repository called from ./lib/ruby_event_store/spec/rom/event_repository_lint.rb:26
```

and

```
     Sequel::DatabaseError:
       SQLite3::SQLException: near "DESC": syntax error
     Shared Example Group: :event_repository called from ./lib/ruby_event_store/spec/rom/event_repository_lint.rb:26
```